research/minigo/minigo.py

```diff
@@ -79,41 +79,17 @@ def bootstrap(estimator_model_dir, trained_models_dir, params):
   dualnet.export_model(estimator_model_dir, bootstrap_model_path)
 
 
-def selfplay(model_name, trained_models_dir, selfplay_dir, holdout_dir, sgf_dir,
-             params):
+def selfplay(selfplay_dirs, selfplay_model, params):
   """Perform selfplay with a specific model.
 
   Args:
-    model_name: The name of the model used for selfplay.
-    trained_models_dir: The path to the model files.
-    selfplay_dir: Where to write the games. Set as 'base_dir/data/selfplay/'.
-    holdout_dir: Where to write the holdout data. Set as
-      'base_dir/data/holdout/'.
-    sgf_dir: Where to write the sgf (Smart Game Format) files. Set as
-      'base_dir/sgf/'.
+    selfplay_dirs: A dict to specify the directories used in selfplay.
+    selfplay_model: The actual Dualnet runner for selfplay.
     params: An object of hyperparameters for the model.
```

Comment:
Using object as type is not ideal for a data entity. If there are specific fields, then we should create a data entity class for it.

Reply:
params is an instance of class MiniGoParams, like params = model_params.MiniGoParams(). Not quite sure about "a data entity class" you mentioned. Could you explain more? Thanks!

Reply:
Then the type of params is not just a object, but MiniGoParams, the comment should be updated accordingly.